src/textual/app.py

```diff
@@ -562,7 +562,20 @@ def __init__(
         """Number of lines to scroll in the Y direction with wheel or trackpad."""
 
         self._installed_screens: dict[str, Screen | Callable[[], Screen]] = {}
+        for var_name, dct in (("SCREENS", self.SCREENS), ("MODES", self.MODES)):
+            for screen_name, v in dct.items():
+                if (
+                    isinstance(v, Screen)
```

Comment:
I think we can simplify a little. The only legal types are a `str` or anything callable.

Maybe this?

```python
if not (isinstance(screen_name, str) or callable(screen_name)):
```

Reply:
I'm going to turn this into two checks: one that raises a `TypeError` for something wrong, and possibly a `ValueError` if it's an *instance* of `Screen` that has a nicer error message (rather than `x is the wrong type`) -- I don't want you guys to get bug reports complaining that instances no longer work!